### PR TITLE
Fufang/fix logging override

### PR DESCRIPTION
**Pull request by fuhuifang**
Update logging to address https://github.com/Azure/confidential-ml-utils/issues/56

### CODE REVIEW

File: src/confidential_ml_utils/logging.py [hunk 1]
```diff
@@ -54,17 +54,31 @@ class ConfidentialLogger(logging.getLoggerClass()):  # type: ignore
...
+            exc_info=exc_info,
+            extra={"prefix": p},
+            stack_info=stack_info,
+            stacklevel=stacklevel,
         )
 
-    def debug(
-        self, msg: str, category: DataCategory = DataCategory.PRIVATE, *args, **kwargs
-    ):
+    def debug(self, msg: str, category=DataCategory.PRIVATE, *args, **kwargs):
```
Comment (danrsc) [inline src/confidential_ml_utils/logging.py:81]: I think the category arg need to move after *args here and for info, warn, etc. Not sure what the superclass signature looks like

File: src/confidential_ml_utils/logging.py [hunk 2]
```diff
@@ -54,17 +54,41 @@ class ConfidentialLogger(logging.getLoggerClass()):  # type: ignore
...
             p = get_prefix()
-        super(ConfidentialLogger, self)._log(
-            level, msg, args, extra={"prefix": p}, **kwargs
-        )
-
-    def debug(
-        self, msg: str, category: DataCategory = DataCategory.PRIVATE, *args, **kwargs
-    ):
+        if sys.version_info[1] <= 7:
+            super(ConfidentialLogger, self)._log(
```
Comment (danrsc) [inline src/confidential_ml_utils/logging.py:72]: One thing you could do to handle future changes to the superclass is use inspect.signature.bind. Might be overkill, just a thought

**Review by danrsc — COMMENTED** [review]
One overall suggestion - could we add some unit tests that explicitly check that non-category aware logging works?

File: src/confidential_ml_utils/logging.py [hunk 3]
```diff
@@ -54,17 +54,41 @@ class ConfidentialLogger(logging.getLoggerClass()):  # type: ignore
...
+                level=level,
+                msg=msg,
+                args=args,
+                exc_info=exc_info,
+                extra={"prefix": p},
+                stack_info=stack_info,
+                stacklevel=stacklevel,
+            )
+
+    def debug(self, msg: str, *args, category=DataCategory.PRIVATE, **kwargs):
```
Comment (danrsc) [inline src/confidential_ml_utils/logging.py:91]: If the superclass passes through category via kwargs, could we just eliminate the debug, info, etc. overrides? We can probably make this as simple as just overriding _log

**Review by dkmiller — CHANGES_REQUESTED** [review]
Needs a version bump, but otherwise looks good to me.

**Review by dkmiller — APPROVED** [review]

**Pull request merged**